sdk/personalizer/azure-ai-personalizer/src/main/java/com/azure/ai/personalizer/PersonalizerAdministrationClientBuilder.java

```diff
@@ -0,0 +1,242 @@
+    AzureKeyCredentialTrait<PersonalizerAdministrationClientBuilder>,
+    ConfigurationTrait<PersonalizerAdministrationClientBuilder>,
+    EndpointTrait<PersonalizerAdministrationClientBuilder>,
+    HttpTrait<PersonalizerAdministrationClientBuilder>,
+    TokenCredentialTrait<PersonalizerAdministrationClientBuilder> {
+    private ClientLogger logger;
+
+    protected void setLogger(ClientLogger logger) {
+        this.logger = logger;
+    }
+
+    private final List<HttpPipelinePolicy> perCallPolicies = new ArrayList<>();
+    private final List<HttpPipelinePolicy> perRetryPolicies = new ArrayList<>();
+
+    private ClientOptions clientOptions;
+    private String endpoint;
+    private AzureKeyCredential azureKeyCredential;
+    private HttpClient httpClient;
+    private HttpLogOptions httpLogOptions;
+    private HttpPipeline httpPipeline;
+    private Configuration configuration;
+    private RetryPolicy retryPolicy;
+    private RetryOptions retryOptions;
+    private TokenCredential tokenCredential;
+    private PersonalizerServiceVersion version;
+    private PersonalizerAudience audience;
+
+    PersonalizerClientV1Preview3Impl getService() {
+        // Endpoint cannot be null, which is required in request authentication
+        Objects.requireNonNull(endpoint, "'Endpoint' is required and can not be null.");
+        if (audience == null) {
+            audience = PersonalizerAudience.AZURE_RESOURCE_MANAGER_PUBLIC_CLOUD;
+        }
+        // Global Env configuration store
+        final Configuration buildConfiguration = (configuration == null)
+            ? Configuration.getGlobalConfiguration().clone() : configuration;
+
+        // Service Version
+        final PersonalizerServiceVersion serviceVersion =
+            version != null ? version : PersonalizerServiceVersion.getLatest();
+
+        HttpPipeline pipeline = httpPipeline;
+        // Create a default Pipeline if it is not given
+        if (pipeline == null) {
+            pipeline = Utility.buildHttpPipeline(
+                clientOptions,
+                httpLogOptions,
+                buildConfiguration,
+                retryPolicy,
+                retryOptions,
+                azureKeyCredential,
+                tokenCredential,
+                audience,
+                perCallPolicies,
+                perRetryPolicies,
+                httpClient);
+        }
+
+        return new PersonalizerClientV1Preview3ImplBuilder()
+            .endpoint(endpoint)
+            .pipeline(pipeline)
+            .buildClient();
+    }
+
+    @Override
+    public PersonalizerAdministrationClientBuilder endpoint(String endpoint) {
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL.", ex));
+        }
+
+        if (endpoint.endsWith("/")) {
```

Comment:
line 114 - line 118 should not be needed if any refernce to this endpoint is appended using URL builder

Reply:
The endpoint value is passed by the customer. So I find value in scrubbing it. Also, the endpoint passed through to the service call after it is scrubbed here. So leaving in place.